src/jit/morph.cpp

```diff
@@ -10772,6 +10778,9 @@ GenTreePtr Compiler::fgMorphCopyBlock(GenTreePtr tree)
             rhs             = fgMorphBlockOperand(rhs, asgType, blockWidth, false /*!isDest*/);
             asg->gtOp.gtOp2 = rhs;
 
+            // Propogate flags up
+            asg->gtFlags |= (rhs->gtFlags & GTF_ALL_EFFECT);
```

Comment:
There is another place in `fgMorphCopyBlock` where we assign `asg->gtOp.gtOp2 = rhs;`
Do we need to updates flags there as well?

Also, it looks like we protect against 'missed flag on a tree', but we do not protect against extra flag, if some side effects were deleted from children. However, it is not important for now.

Reply:
> we do not protect against extra flag, if some side effects were deleted from children. However, it is not important for now.

I believe that this is a known issue, i.e. we don't re-analyze the entire tree when a side effect is deleted.